.github/workflows/test.yml

```diff
@@ -1,15 +1,60 @@
-name: End 2 End Tests (Spaceflights Quickstart)
+name: Unit and E2E Tests workflow (Spaceflights Quickstart)
 
 on: # workflow_dispatch
   push:
-     branches:
-       - develop
+    branches:
+      - develop
+    paths: 
+      - "**.py"
+      - "!docs/**"
+      - ".github/workflows/test.yml"
```

Comment:
Maybe also add `tests/e2e/` path to trigger here?

Reply:
I have a mixed feelings restricting triggers to run only on modified paths, you can never be sure if those were not run due to the condition or if there's misconfiguration of some sort. If you're changing only documentation you can always add `[no ci]` to the commit message that will skip the actions execution, but you do this consciously. 

Currently the costs are not such big of a deal to worry about that, and there's not much money saved: 
* we don't pay for github runners
* currently kfp cluster has a fixed size, so we pay even if it doesn't run tests

Reply:
how about only exclusion then?
path_exclude:
"docs/**"

Reply:
To be clear, according to documentation:
```
When all the path names match patterns in paths-ignore, the workflow will not run. If any path names do not match patterns in paths-ignore, even if some path names match the patterns, the workflow will run.
```